jsonrpc/ipc_all.go

```diff
@@ -0,0 +1,62 @@
+//go:build !windows
+// +build !windows
+
+package jsonrpc
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"path/filepath"
+)
+
+const (
+	// http://man7.org/linux/man-pages/man7/unix.7.html
+	maxUnixPathSize = int(108)
+)
+
+var errPathTooLong = errors.New("path too long")
+
+// createListener creates a Unix domain socket at the given path and sets proper permissions.
+func createListener(endpoint string) (net.Listener, error) {
+	var (
+		l   net.Listener
+		err error
+	)
+	// path + terminator
+	if len(endpoint)+1 > maxUnixPathSize {
+		return nil, errPathTooLong
+	}
+	err = preparePath(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	l, err = net.Listen("unix", endpoint)
+	if err != nil {
+		return nil, err
+	}
+	// Set permissions for the socket file to read and write for the owner only (0o600)
+	err = os.Chmod(endpoint, 0o600) //nolint:gomnd
+	return l, err
+}
+
+func UnixDial(ctx context.Context, endpoint string) (net.Conn, error) {
+	return new(net.Dialer).DialContext(ctx, "unix", endpoint)
+}
+
+// preparePath ensures the existence of directories leading to provided path
+// and removes existing file if any.
+func preparePath(path string) error {
+	var err error
+	err = os.MkdirAll(filepath.Dir(path), 0o751) //nolint:gomnd
```

Comment:
What is the reasoning for `0o751`?

Reply:
`751` is a stricter version of `755` which is a common default for directories, eg. ubuntu default homes are of such mode. I wasn't sure between `751` and `750` but decided to go with the first one since it's also used in `go-etherum`.

Reply:
Thanks for the explanation, sounds good to me. Can you please summarize the reasoning in a comment for future reference? Extracting it out to a constant instead of using `nolint` would be nice as well.